src/catalog/database_catalog.cpp

```diff
@@ -1330,6 +1330,38 @@ std::vector<std::pair<common::ManagedPointer<storage::index::Index>, const Index
   return index_objects;
 }
 
+bool DatabaseCatalog::CopyPR(common::ManagedPointer<transaction::TransactionContext> txn,
```

Comment:
This is logic for materializing the tuple from the index scan? It doesn't belong in the `Catalog`. This should be in `StorageInterface`, not called into from there.

Reply:
Now these operations are moved to the storage interface and stashed in the initialization